staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -505,14 +505,12 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 
 // Resync will send a sync event for each item
 func (f *DeltaFIFO) Resync() error {
-	var keys []string
-	func() {
-		f.lock.RLock()
-		defer f.lock.RUnlock()
-		keys = f.knownObjects.ListKeys()
-	}()
+	f.lock.Lock()
```

Comment:
Yes, although (as best I can tell), Resync is only ever called during an actual resync, and Replace is only ever called if the watch closes and the reflector restarts ListAndWatch.

If this PR is ultimately approved, it makes the code in #40759 a lot cleaner